### PR TITLE
Convert examples to use op namespace

### LINKED ISSUE #691: main CI is failing
I guess `numpy.asscalar` stopped working?   
https://github.com/illinois-ceesd/mirgecom/runs/7033960856?check_suite_focus=true#step:4:58  

fixed by #689
changed files: `simutil.py`

### PR DESCRIPTION
**Pull request by MTCam**
This is a follow-on to #687 and #688.  It moves the examples from `EagerDGDiscretization` --> `DiscretizationCollection`.

Plus a couple of out-of-band changes:
- `discretization.py`: Update discr coll util to use `default_node_factory` util from `meshmode` 
- `simutil.py`: Resolve deprecated `asscalar@numpy`, fixes #691 

 
**Questions for the review** @lukeolson:
- [x] Is the scope and purpose of the PR clear?
  - [x] The PR should have a description.
  - [x] The PR should have a guide if needed (e.g., an ordering).
- [ ] ~~Is every top-level method and class documented? Are things that should be documented actually so?~~
- [ ] ~~Does the implementation do what the docstring claims?~~
- [x] Is the interface understandable? (I.e. can someone figure out what stuff does?) Is it well-defined?
- [x] Is everything that is implemented covered by tests?
- [x] Do you see any immediate risks or performance disadvantages with the design? Example: what do interface normals attach to?

### CODE REVIEW

File: mirgecom/discretization.py [hunk 1]
```diff
@@ -45,18 +45,19 @@ def create_discretization_collection(actx, mesh, order, *, mpi_communicator=None
     """Create and return a grudge DG discretization object."""
     from grudge.dof_desc import DISCR_TAG_BASE, DISCR_TAG_QUAD
     from grudge.eager import EagerDGDiscretization
-    from meshmode.discretization.poly_element import \
-        QuadratureSimplexGroupFactory, \
-        PolynomialRecursiveNodesGroupFactory
+    from meshmode.discretization.poly_element import (
+        QuadratureSimplexGroupFactory,
+        default_simplex_group_factory
```
Comment (lukeolson) [inline mirgecom/discretization.py:50]: Why is this change needed? it just calls PolynomialRecursiveNodesGroupFactory anyway...
https://github.com/inducer/meshmode/blob/main/meshmode/discretization/poly_element.py#L952

Comment (MTCam) [reply mirgecom/discretization.py:50]: This one is more robust in that it will continue to work even if the TPL `recursivenodes` is not installed.   Older installations will not have this, and reproducing an older run will need this more robust version to work.

Comment (lukeolson) [reply mirgecom/discretization.py:50]: ok, import wrapper in case `recursivenodes` is unavailable.
